src/reference/asciidoc/redis.adoc

```diff
@@ -769,6 +769,94 @@ IMPORTANT: The `task-executor` has to be configured with more than one thread fo
 The `errorChannel` can be used to process those errors, to avoid restarts, but it preferable to not expose your application to the possible deadlock situation.
 See Spring Framework https://docs.spring.io/spring/docs/current/spring-framework-reference/integration.html#scheduling-task-executor-types[Reference Manual] for possible `TaskExecutor` implementations.
 
+[[redis-stream-outbound]]
+=== Redis Stream Outbound Channel Adapter
+
+Spring Integration 5.4 introduced Reactive Redis Stream outbound channel adapter to write Message payload into Redis stream.
+Outbound Channel adapter uses `ReactiveStreamOperations.add(...)` to send `Message` to stream.
+The following example shows how to use Java Configuration and Service class for Redis Stream Outbound Channel Adapter.
+
+====
+[source,java]
+----
+@Component
+public class MyClass {
+
+    @Bean
+    @ServiceActivator(inputChannel = "messageChannel")
+    public ReactiveRedisStreamMessageHandler reactiveValidatorMessageHandler(ReactiveRedisConnectionFactory reactiveRedisConnectionFactory) {
+            ReactiveRedisStreamMessageHandler reactiveStreamMessageHandler =
+                new ReactiveRedisStreamMessageHandler(reactiveRedisConnectionFactory, "myStreamKey"); <1>
+            reactiveRedisStreamMessageHandler.setSerializationContext(serializationContext); <2>
+            reactiveRedisStreamMessageHandler.setHashMapper(hashMapper); <3>
+            reactiveRedisStreamMessageHandler.setExtractPayload(true); <4>
+     }
+}
+----
+<1> Construct an instance of `ReactiveRedisStreamMessageHandler` using `ReactiveRedisConnectionFactory` and expression for `StreamKey`.
+<2> Set `RedisSerializationContext` which provides SerializationPairs for key, value, hash-key (field), hash-value and String serialization and deserialization.
+<3> Set `HashMapper` which provides contract between Java types and Redis hashes/maps.
+<4> If 'true', channel adapter will extract payload from `Message` and then send it to Stream otherwise entire message is sent.
+It defaults to `true`.
+====
+
+[[redis-stream-inbound]]
+=== Redis Stream Inbound Channel Adapter
+
+Spring Integration 5.4 introduced the Reactive Stream inbound channel adapter for reading messages from a Redis Stream.
+Inbound channel adapter uses `StreamReceiver.receive(...)` or `StreamReceiver.receiveAutoAck` based on auto acknowledgement flag to read `Message` from redis stream.
```

Comment:
Yeah... It doesn't read a `Message`, but rather `Record` and then build a `Message` with an appropriate headers.